### DISCUSSION TITLE
Detecting if a type map is registered

### DISCUSSION THREAD

**Discussion by replaysMike** [LuckyPennySoftware/AutoMapper#3603] [NA]
It seems there are a few entry points to get type maps for a specified type pair, such as:

```
var typePair = new TypePair(type1, type2);
Mapper.ConfigurationProvider.FindTypeMapFor(typePair);
Mapper.ConfigurationProvider.ResolveTypeMap(typePair);
```

However it returns null for variants of the type pair, such as arrays/collections/dictionaries etc. There must be some internal handling done before calling these methods since it can definitely map the types, and it's not exposed by the api. Is there any other way to determine if Automapper will map a type besides catching mapping exceptions during a call to `.Map<Type>()`?

The reason why isn't really the question here, but more as to if it can be done.

**Comment by lbargaoanu** [comment]
https://github.com/AutoMapper/AutoMapper/blob/9f2f16067ab201a5a8b9bc982f3a37e8790da7a0/src/AutoMapper/Configuration/MapperConfiguration.cs#L133
